test/namenode/NameNodeTest.cc

```diff
@@ -120,6 +120,22 @@ namespace {
 		zk->exists("/fileSystem/old_name", exist, error_code);
 		ASSERT_EQ(false, exist);
 	}
+
+	TEST_F(NamenodeTest, previousBlockComplete){
+		int error;
+		u_int64_t block_id;
+		util::generate_uuid(block_id);
+		LOG(INFO) << "Previous block_id is " << block_id;
+		ASSERT_EQ(false, client->previousBlockComplete(block_id));
+		/* mock the directory */
+		zk->create("/block_locations", ZKWrapper::EMPTY_VECTOR, error);
+		zk->create("/block_locations/"+std::to_string(block_id), ZKWrapper::EMPTY_VECTOR, error);
+		ASSERT_EQ(false, client->previousBlockComplete(block_id));
```

Comment:
previousBlockComplete should not be a public method. So, we should only be able to test it through addBlock. Other people might have a different opinion about this, though.